resources/serverless/templates/virtualservice.yaml

```diff
@@ -0,0 +1,36 @@
+{{- if .Values.dockerRegistry.enableInternal }}
+apiVersion: networking.istio.io/v1beta1
+kind: VirtualService
+metadata:
+  name: {{ template "registry-fullname" . }}
+  namespace: kyma-system
+  labels:
+  {{- include "tplValue" ( dict "value" .Values.global.commonLabels "context" . ) | nindent 4 }}
+spec:
+  gateways:
+    - kyma-system/kyma-gateway
+  hosts:
+    - registry.{{ .Values.global.ingress.domainName }} # docker-registry.{DOMAIN}
+  http:
+    - corsPolicy:
+        allowHeaders:
+          - Authorization
+          - Content-Type
+          - '*'
+        allowMethods:
+          - GET
+          - HEAD
+          - POST
+          - PUT
+          - DELETE
+        allowOrigins:
+          - regex: .*
+      match:
```

Comment:
Previously, an Oathkeeper rule was configured to only match `GET` and `HEAD` requests on any path. Now we allow all methods. A `method` parameter should be added and perhaps another HTTPMatchRequest object. See: https://istio.io/v1.9/docs/reference/config/networking/virtual-service/#HTTPMatchRequest